xarray/backends/api.py

```diff
@@ -1596,27 +1600,67 @@ def open_mfdataset(
 
     open_kwargs = dict(engine=engine, chunks=chunks or {}, **kwargs)
 
-    if parallel:
+    if parallel is True:
+        warnings.warn(
+            "Passing ``parallel=True`` is deprecated, instead please pass ``parallel='dask'`` explicitly",
+            PendingDeprecationWarning,
+            stacklevel=2,
+        )
+        parallel = "dask"
+
+    if parallel == "dask":
         import dask
 
         # wrap the open_dataset, getattr, and preprocess with delayed
         open_ = dask.delayed(open_dataset)
         getattr_ = dask.delayed(getattr)
         if preprocess is not None:
             preprocess = dask.delayed(preprocess)
-    else:
+    elif parallel == "lithops":
+        import lithops
+
+        # TODO use RetryingFunctionExecutor instead?
+        fn_exec = lithops.FunctionExecutor()
+
+        # lithops doesn't have a delayed primitive
+        open_ = open_dataset
+        # TODO I don't know how best to chain this with the getattr
+        # getattr_ = getattr
+    elif parallel is False:
         open_ = open_dataset
         getattr_ = getattr
+    else:
+        raise ValueError(
+            f"{parallel} is an invalid option for the keyword argument ``parallel``"
+        )
 
-    datasets = [open_(p, **open_kwargs) for p in paths1d]
-    closers = [getattr_(ds, "_close") for ds in datasets]
-    if preprocess is not None:
-        datasets = [preprocess(ds) for ds in datasets]
+    if parallel == "dask":
+        datasets = [open_(p, **open_kwargs) for p in paths1d]
+        closers = [getattr_(ds, "_close") for ds in datasets]
+        if preprocess is not None:
+            datasets = [preprocess(ds) for ds in datasets]
 
-    if parallel:
         # calling compute here will return the datasets/file_objs lists,
         # the underlying datasets will still be stored as dask arrays
         datasets, closers = dask.compute(datasets, closers)
+    elif parallel == "lithops":
+
+        def generate_lazy_ds(path):
+            # allows passing the open_dataset function to lithops without evaluating it
+            ds = open_(path, **kwargs)
+            return ds
+
+        futures = fn_exec.map(generate_lazy_ds, paths1d)
+
+        # wait for all the serverless workers to finish, and send their resulting lazy datasets back to the client
+        # TODO do we need download_results?
+        completed_futures, _ = fn_exec.wait(futures, download_results=True)
+        datasets = completed_futures.get_result()
```

Comment:
I wonder if we can find an abstraction that works for both this (which is kinda like `concurrent.futures`' pool executors) and `dask`.

For example, maybe we can use `functools.partial` to mimic `dask.delayed`. The result would be a bunch of function objects without any parameters, which would then be evaluated in `fn_exec.map` using `operator.call`.